unic-locale-impl/src/extensions/unicode.rs

```diff
@@ -81,14 +145,100 @@ impl UnicodeExtensionList {
+    ///
+    /// Returns `true` if keyword was included in the `UnicodeExtensionList`
+    /// before removal.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-u-ca-buddhist".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.unicode.remove_keyword("ca")
+    ///                .expect("Removing tag failed."),
+    ///            true);
+    ///
+    /// assert_eq!(lo.to_string(), "en-US");
+    /// ```
+    pub fn remove_keyword(&mut self, key: &str) -> Result<bool, LocaleError> {
+        Ok(self.keywords.remove(&parse_key(key)?).is_some())
+    }
+
+    /// Returns `true` if attribute is included in the `UnicodeExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-u-foo".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.unicode.get_attribute("foo")
+    ///                .expect("Getting attribute failed."),
+    ///            true);
+    /// ```
+    pub fn get_attribute(&self, attribute: &str) -> Result<bool, LocaleError> {
+        Ok(self.attributes.contains(&parse_attribute(attribute)?))
+    }
+
+    /// Sets an attribute on the `UnicodeExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// lo.extensions.unicode.set_attribute("foo")
+    ///     .expect("Setting attribute failed.");
+    ///
+    /// assert_eq!(lo.to_string(), "en-US-u-foo");
+    /// ```
+    pub fn set_attribute(&mut self, attribute: &str) -> Result<(), LocaleError> {
+        let attribute = parse_attribute(attribute)?;
+        if let Err(idx) = self.attributes.binary_search(&attribute) {
+            self.attributes.insert(idx, attribute);
         }
         Ok(())
     }
 
+    /// Removes an attribute from the `UnicodeExtensionList`.
+    ///
+    /// Returns `true` if attribute was included in the `UnicodeExtensionList`
+    /// before removal.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-u-foo".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.unicode.remove_attribute("foo")
+    ///                .expect("Removing attribute failed."),
+    ///            true);
+    ///
+    /// assert_eq!(lo.to_string(), "en-US");
+    /// ```
+    pub fn remove_attribute(&mut self, attribute: &str) -> Result<bool, LocaleError> {
+        let attribute = parse_attribute(attribute)?;
+        match self.attributes.binary_search(&attribute) {
+            Ok(idx) => {
+                self.attributes.remove(idx);
+                Ok(true)
+            },
+            Err(_) => Ok(false)
+        }
+    }
+
```

Comment:
`get_attribute_keys` and `clear_attributes`/`clear_keywords`.

Reply:
Leaving `get_attribute_keys` for separate commit too.